frontend-desktop/src/main.rs

```diff
@@ -145,24 +174,51 @@ fn main() {
         let websocket_client = websocket_client.clone();
         let config = config.clone();
         move || {
-            let json_payload = json!({"name": "RustCoffee", "type": "join"});
-            send(&websocket_client, json_payload, "coffeeRequest");
+            let json_payload = json!({"name": &config.username, "type": "join"});
+            match websocket_client.as_ref() {
+                Some(x) => send(x, json_payload, "coffeeRequest"),
+                None => (),
+            }
         }
     });
     ui.on_go({
         let websocket_client = websocket_client.clone();
         let config = config.clone();
         move || {
-            let json_payload = json!({"name": "RustCoffee", "type": "start"});
-            send(&websocket_client, json_payload, "coffeeRequest");
+
+            let json_payload = json!({"name": &config.username, "type": "start"});
+            match websocket_client.as_ref() {
+                Some(x) => send(x, json_payload, "coffeeRequest"),
+                None => (),
+            }
+        }
+    });
+    ui.on_save_and_reconnect({
+        let main_window_weak = ui.as_weak();
+        let config = config.clone();
+        move |name, address| {
+            let new_config = CoffeeCallerConfig { username: name.as_str().into(), server: address.as_str().into() };
+            let toml = toml::to_string(&new_config).unwrap();
+            let file = OpenOptions::new().write(true).create(true).truncate(true).open("config.toml");
```

Comment:
This file has as suffix, but in line 128 a different config is read? Is that on purpose?

Reply:
The reading thing has some magic, I don't know how it works and I was super confused too, but it doesn't "need" a file type ending. It just chooses that automatically, no idea how. I'll check if I can add it to the reading part.